src/btree/bt_read.c

```diff
@@ -194,7 +194,7 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
     page_flags = WT_DATA_IN_ITEM(&tmp) ? WT_PAGE_DISK_ALLOC : WT_PAGE_DISK_MAPPED;
     if (LF_ISSET(WT_READ_IGNORE_CACHE_SIZE))
         FLD_SET(page_flags, WT_PAGE_EVICT_NO_PROGRESS);
-    WT_ERR(__wt_page_inmem(session, ref, tmp.data, page_flags, &notused, &prepare));
+    WT_ERR(__wt_page_inmem(session, ref, tmp.data, tmp.size, page_flags, &notused, &prepare));
```

Comment:
Is this always the right size? At the moment, the correct size is used for pages that were compressed, and I haven't done the work to understand why, so I'm asking you because I'm lazy:
Will this be correct for cases where the page was block compressed and not block compressed?

Reply:
Yes, I read the code and the size seemed correct. I will spend some more time on this.

Reply:
To answer @agorrod's question:

> the correct size is used for pages that were compressed, and I haven't done the work to understand why, so I'm asking you because I'm lazy

When we read a compressed page, we read it into a buffer that matches the on-disk size. Then we read the in-memory size from the page header (which isn't compressed). We allocate a new buffer using this size and decompress the page into that new buffer. So the we wind up with a buffer of the correct size.